### PR TITLE
added server functions to fetch extra initial data (plugin side)

### PR DESCRIPTION
**Pull request by arcadianCdr**
Counterpart of https://github.com/porn-vault/porn-vault/pull/1253

### CODE REVIEW

File: types/studio.ts [hunk 1]
```diff
@@ -30,4 +31,28 @@ export type StudioOutput = Partial<FullStudioOutput>;
...
+
+  const initialData: StudioOutput = {
+    name: studio.name,
+    description: studio.description || undefined,
+    addedOn: studio.addedOn.valueOf(),
+    favorite: studio.favorite,
+    bookmark: studio.bookmark || undefined,
+    aliases: studio.aliases,
+    labels: (await ctx.$getLabels()).map((l) => l.name),
+    parent: (await ctx.$getParents()).map((p) => p.name)?.[0],
```
Comment (leadwolf) [inline types/studio.ts:53]: real nitpicking, but you don't need to map if you're only going to use the first element
```suggestion
    parent: (await ctx.$getParents())?.[0]?.name,
```

Comment (arcadianCdr) [reply types/studio.ts]: Removed those ;-)

**Review by leadwolf — APPROVED** [review]

File: context.ts [hunk 2]
```diff
@@ -173,7 +179,13 @@ function createPluginLogger(name: string): winston.Logger {
   });
 }
 
-const context: Context = {
+function warnServerFunc(functionName: string) {
+  console.error(
```
Comment (arcadianCdr) [inline context.ts:183]: I wanted to throw a new Error, but the exception seems to be swallowed and never reaches the test. Do you know why?

Comment (arcadianCdr) [reply context.ts]: Was a problem on my side... Updated to throw...

**Pull request merged**